### ISSUE TITLE
Test failure testMetadataCleanupRecovery

### ISSUE THREAD

**Issue by fpj** [pravega/pravega#3127]
**Problem description**
Observed this test failure in our internal jenkins:

```
io.pravega.segmentstore.server.containers.StreamSegmentContainerTests.testMetadataCleanupRecovery
```
Stack trace:

```
Error Message

java.lang.IllegalStateException: Expected the service MetadataCleanupContainer [FAILED] to be TERMINATED, but the service has FAILED
Stacktrace

java.lang.IllegalStateException: Expected the service MetadataCleanupContainer [FAILED] to be TERMINATED, but the service has FAILED
	at com.google.common.util.concurrent.AbstractService.checkCurrentState(AbstractService.java:328)
	at com.google.common.util.concurrent.AbstractService.awaitTerminated(AbstractService.java:293)
	at io.pravega.segmentstore.server.containers.StreamSegmentContainerTests.testMetadataCleanupRecovery(StreamSegmentContainerTests.java:1276)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:298)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:292)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.util.concurrent.TimeoutException: Timeout expired while waiting for the Service to shut down.
	at io.pravega.common.concurrent.AbstractThreadPoolService.lambda$doStop$1(AbstractThreadPoolService.java:115)
	at java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:760)
	at java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:736)
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:474)
	at java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:1962)
	at io.pravega.common.concurrent.Futures.lambda$delayedFuture$19(Futures.java:516)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	... 1 more
```

**Problem location**
Segment store

**Suggestions for an improvement**
Analyze the problem and fix the test failure.

**Comment by andreipaduroiu** [comment]
I have changed the underlying code significantly with PR #3206 . Let's see if this still happens.

**Comment by andreipaduroiu** [comment]
I ran this test in a loop in a VM using 1 CPU with 50% execution cap. When targeting branch `r0.4`, this fails consistently after a few runs. When targeting the current `master`, this does not fail (waited 200 runs before stopping the execution).

If this happens again, we can reopen the issue.

**Comment by derekm** [comment]
@andreipaduroiu -- Seen in a recent failed Actions build:
```
io.pravega.segmentstore.server.containers.StreamSegmentContainerTests > testMetadataCleanupRecovery FAILED
    java.lang.IllegalStateException: Expected the service MetadataCleanupContainer [FAILED] to be TERMINATED, but the service has FAILED
        at com.google.common.util.concurrent.AbstractService.checkCurrentState(AbstractService.java:366)
        at com.google.common.util.concurrent.AbstractService.awaitTerminated(AbstractService.java:329)
        at io.pravega.segmentstore.server.containers.StreamSegmentContainerTests.testMetadataCleanupRecovery(StreamSegmentContainerTests.java:1513)

        Caused by:
        java.util.concurrent.CompletionException: java.lang.IllegalStateException: Service expected to be NEW but was TERMINATED.
            at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:314)
            at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:319)
            at java.base/java.util.concurrent.CompletableFuture$UniCompose.tryFire(CompletableFuture.java:1081)
            at java.base/java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:478)
            at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
            at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
            at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
            at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
            at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
            at java.base/java.lang.Thread.run(Thread.java:834)

            Caused by:
            java.lang.IllegalStateException: Service expected to be NEW but was TERMINATED.
                at com.google.common.base.Preconditions.checkState(Preconditions.java:824)
                at io.pravega.common.concurrent.Services.startAsync(Services.java:39)
                at io.pravega.segmentstore.server.containers.StreamSegmentContainer.startSecondaryServicesAsync(StreamSegmentContainer.java:311)
                at io.pravega.segmentstore.server.containers.StreamSegmentContainer.lambda$startWhenDurableLogOnline$5(StreamSegmentContainer.java:281)
                at java.base/java.util.concurrent.CompletableFuture$UniCompose.tryFire(CompletableFuture.java:1072)
                ... 7 more
```

**Comment by andreipaduroiu** [comment]
Fixed with #5454
